### DISCUSSION TITLE
How to get the block.timestamp and data with publicClient.getTransaction(hash)?

### DISCUSSION THREAD

**Discussion by Steveantor** [wevm/viem#1069] [General]
It seems like `publicClient.getTransaction(hash)` doesn't return _block.timestamp_ as per the docs,
but `ethers.getTransaction()` in fact does.

I guess the `txinfo.input` is `txinfo.data` (ethers) (unsure about this one)

**Comment by Falci** [comment]
For cross reference, Alchemy SDK seems to have the same issue: alchemyplatform/alchemy-sdk-js/issues/366

**Reply by jxom** [reply]
`getTransaction` returns the transaction only, with no block data (retrieving this data is another JSON-RPC request).

To get the block timestamp, you’ll need to make a call to `getBlock` with the block number on the transaction.
